src/app/app.component.html

```diff
@@ -44,6 +44,12 @@
         <plh-main-header
           [style.display]="headerConfig().hidden ? 'none' : 'block'"
         ></plh-main-header>
+        <!-- Add spacer element when header is hidden to account for safe area (native status bar, notch etc.) -->
+        <div
+          class="safe-area-spacer"
+          [style.display]="headerConfig().hidden ? 'block' : 'none'"
+          [style.height]="'var(--ion-safe-area-top, 0)'"
+        ></div>
```

Comment:
nit(non-blocking): I think it might be a little cleaner to bundle all this logic within the `plh-main-header` component so that it can handle itself either rendering content when config enables, or just placeholder safe-area padding when not. I think the `ion-header` component might do this by default (essentially what you are recreating when hiding it), so it might be possible always render and just show/hide inner content based on config and still get the correct padding adjustments without styling

Reply:
That makes sense. I did consider this approach but thought that handling the colour of the toolbar element might be tricky. In fact I think it makes sense to simply set the colour to "none" in the case that the header is hidden at the app config level. I've made an alternative PR with this approach here: #2701